tests/pytest/test_solvers.py

```diff
@@ -0,0 +1,367 @@
+    # scale factor determines how large the off diagonal elements are
+    M = M + scale * np.random.randn(size, size)
+    # sym factor determines how symmetric the thing is, 1.0=completely , 0.0 not at all
+    M = (M.T * (sym) + M) / 2.0
+    return M
+
+
+class DSProblemSimulate(object):
+    "Provide the interface of an engine, around an actual matrix stored in memory"
+
+    def __init__(self, size, sep=1.0, scale=0.0001, sym=1.0):
+        self.size = size
+        self.A = _diag_dom_sym_mat(size, sep=sep, scale=scale, sym=sym)
+
+    def compute_products(self, X):
+        return self.A.dot(X)
+
+    def precondition_residual(self, R, shift):
+        return R / (shift - np.diag(self.A))
+
+    def approximate_eigvector(self, Ax, Vss):
+        return np.dot(Ax, Vss)
+
+    def residual(self, V, x, ss_vector, ss_value):
+        return V - np.dot(x, ss_vector * ss_value)
+
+    def vector_norm(self, V):
+        return np.linalg.norm(V)
+
+    def orthogonalize_subspace(self, old_X, new_X):
+        if isinstance(old_X, list):
+            assert len(old_X) == 0
+            old_x_cols = tuple()
+        else:
+            old_x_cols = tuple(old_X.T)
+        if isinstance(new_X, np.ndarray):
+            new_x_cols = tuple(new_X.T)
+        else:
+            new_x_cols = tuple(new_X)
+
+        X = np.column_stack(old_x_cols + new_x_cols)
+        X, _ = np.linalg.qr(X)
+        return X, X.shape[-1]
+
+    def subspace_project(self, X, Ax):
+        return np.dot(X.T, Ax)
+
+
+class HSProblemSimulate(object):
+    "Provide the interface of an engine, around an actual matrix stored in memory"
+
+    def __init__(self, size, a_sep=1.0, b_sep=0.001, scale=0.0001, a_sym=1.0, b_sym=1.0):
+        self.size = size
+        self.A = _diag_dom_sym_mat(size, sep=a_sep, scale=scale, sym=a_sym)
+        self.B = _diag_dom_sym_mat(size, sep=b_sep, scale=scale, sym=b_sym)
+
+    def compute_products(self, X):
+        Px = np.dot(self.A + self.B, X)
+        Mx = np.dot(self.A - self.B, X)
+        return Px, Mx
+
+    def precondition_residual(self, R, shift):
+        return R / (shift - np.diag(self.A))
+
+    def approximate_eigvector(self, Ax, Vss):
+        return np.dot(Ax, Vss)
+
+    def residual(self, V, x, ss_vector, ss_value):
+        return V - np.dot(x, ss_vector * ss_value)
+
+    def vector_norm(self, V):
+        return np.linalg.norm(V)
+
+    def orthogonalize_subspace(self, old_X, new_X):
+        if isinstance(old_X, list):
+            assert len(old_X) == 0
+            old_x_cols = tuple()
+        else:
+            old_x_cols = tuple(old_X.T)
+        if isinstance(new_X, np.ndarray):
+            new_x_cols = tuple(new_X.T)
+        else:
+            new_x_cols = tuple(new_X)
+
+        X = np.column_stack(old_x_cols + new_x_cols)
+        X, _ = np.linalg.qr(X)
+        return X, X.shape[-1]
+
+    def subspace_project(self, X, Ax):
+        return np.dot(X.T, Ax)
+
+
+@pytest.mark.unittest
+@pytest.mark.solver
```

Comment:
I will also remove these additional marks in a cleanup pass later, unless we want to keep them. 

I just used them to run subsets of the tests